src/hooks/data.tsx

```diff
@@ -173,54 +174,35 @@ export const DataProvider = ({
       });
   }, []);
 
-  // calculate distance if coordinates are available
+  // Update data when location changes to recalculate distances
   useEffect(() => {
-    if (!latitude || !longitude || !data.meetings || data.waitingForData) {
+    if (data.waitingForData || !Object.keys(data.meetings).length) {
       return;
     }
 
-    const distances = Object.fromEntries(
-      settings.distance_options.map(option => [option, []])
-    );
-
-    Object.keys(data.meetings).forEach(slug => {
-      const meeting = data.meetings[slug];
-      if (meeting.latitude && meeting.longitude) {
-        meeting.distance = getDistance(
-          { latitude, longitude },
-          meeting,
-          settings
-        );
-      }
-
-      for (const option of settings.distance_options) {
-        if (meeting.distance && meeting.distance <= option) {
-          (distances[option] as string[]).push(meeting.slug);
-        }
-      }
-
-      data.meetings[slug] = meeting;
-    });
-
-    const distance: Index[] = Object.entries(distances).map(([key, slugs]) => ({
-      key,
-      name: `${key} ${settings.distance_unit}`,
-      slugs,
-    }));
+    const { meetings: meetingsWithDistances, distanceIndex, hasDistance } = calculateDistances(data.meetings);
 
     setData(prevData => ({
       ...prevData,
       capabilities: {
         ...prevData.capabilities,
-        distance: true,
+        distance: hasDistance,
       },
       indexes: {
         ...prevData.indexes,
-        distance,
+        distance: distanceIndex,
       },
-      meetings: data.meetings,
+      meetings: meetingsWithDistances,
     }));
-  }, [latitude, longitude, data.meetings, settings.distance_unit]);
+  }, [calculateDistances, data.waitingForData]);
 
   return <DataContext.Provider value={data}>{children}</DataContext.Provider>;
 };
+
+export const DataProvider = (props: PropsWithChildren<{ google?: string; src?: string; timezone?: string }>) => {
+  return (
+    <LocationProvider>
```

Comment:
thanks so much for working on this! i'm going to approve and merge because this fixes the issue but i think in the next update i might want to move this LocationProvider call to the TsmlUi component unless there's a specific reason not to